neo/rawio/intanrawio.py

```diff
@@ -72,42 +76,106 @@ class IntanRawIO(BaseRawIO):
+            self._raw_data = {}
+            for stream_index, (stream_index_key, stream_datatype) in enumerate(data_dtype.items()):
+                num_channels = channel_number_dict[stream_index_key]
+                file_path = raw_file_paths_dict[stream_index_key]
+                size_in_bytes = file_path.stat().st_size
+                dtype_size = np.dtype(stream_datatype).itemsize
+                n_samples = size_in_bytes // (dtype_size * num_channels)
+                signal_stream_memmap = np.memmap(file_path, dtype=stream_datatype, mode="r", shape=(num_channels, n_samples)).T
+                self._raw_data[stream_index] = signal_stream_memmap
+
+        # for one-file-per-channel we have one memory map / channel stored as a list / neo stream
+        elif self.file_format == "one-file-per-channel":
+            self._raw_data = {}
+            for stream_index, (stream_index_key, stream_datatype) in enumerate(data_dtype.items()):
+                self._raw_data[stream_index] = []
+                num_channels = channel_number_dict[stream_index_key]
+                for channel_index in range(num_channels):
+                    file_path = raw_file_paths_dict[stream_index_key][channel_index]
+                    channel_memmap = np.memmap(file_path, dtype=stream_datatype, mode="r")
+                    self._raw_data[stream_index].append(channel_memmap)
 
-        # memmap raw data with the complicated structured dtype
-        self._raw_data = np.memmap(self.filename, dtype=data_dtype, mode="r", offset=header_size)
 
         # check timestamp continuity
-        timestamp = self._raw_data["timestamp"].flatten()
-        assert np.all(np.diff(timestamp) == 1), "timestamp have gaps"
+        if self.file_format == "header-attached":
+            timestamp = self._raw_data["timestamp"].flatten()
+        # timestamps are always the last stream
+        elif self.file_format == "one-file-per-signal":
+            time_stream_index = max(self._raw_data.keys())
+            timestamp = self._raw_data[time_stream_index]
+        elif self.file_format == "one-file-per-channel":
+            time_stream_index = max(self._raw_data.keys())
+            timestamp = self._raw_data[time_stream_index][0]
+
+        assert np.all(np.diff(timestamp) == 1), (
+            "Timestamp have gaps, this could be due " "to a corrupted file or an inappropriate file merge"
+        )
 
         # signals
         signal_channels = []
         for c, chan_info in enumerate(self._ordered_channels):
             name = chan_info["custom_channel_name"]
             channel_id = chan_info["native_channel_name"]
-            if chan_info["signal_type"] == 20:
-                # exception for temperature
+            if chan_info["signal_type"] == 20 or (self.file_format != "header-attached" and chan_info["signal_type"] == 0):
+                # exception for temperature and for amplifier in non-header attached files
```

Comment:
@ChauhanT 

Good catch for the header info. We hadn't updated this yet for our updates (@h-mayorquin). So you were right that the info was being displayed wrong, but it wasn't hurting anything. You can test this new commit to see if it fixes the header display, but when I tested it locally it worked. :)

Reply:
@h-mayorquin if we decide to allow timestamps in the future, then we will need to fix this solution more robustly then what I just did. (So if you have a brilliant idea for a better fix, just let me know).

Reply:
Add the dtype to `chan_info` in the `read_rhd` function and then use it here:

https://github.com/zm711/python-neo/blob/4283378cadc734b844cbc757d65e39c58d1a93d1/neo/rawio/intanrawio.py#L786-L791

Reply:
smart. I did it for rhs too, because this is a common function for both formats.